pandas/io/pytables.py

```diff
@@ -3916,7 +3911,7 @@ def get_blk_items(mgr, blocks):
             j += 1
 
         self.nan_rep = nan_rep
-        self.data_columns = new_data_columns
+        self.data_columns = [col.name for col in vaxes if col.is_data_indexable]
```

Comment:
i think/hope that something like this is the general case and data_columns can become a property